DocumentationValidator/Program.cs

```diff
@@ -14,11 +15,69 @@ class Program
         static async Task<int> Main(string[] args)
         {
             IList<NuGetLogCode> allLogCodes = GetNuGetLogCodes();
-            List<NuGetLogCode> undocumentedLogCodes = new();
             HttpClient httpClient = new();
+            var cred = args.Length > 0 ? args[0] : "123";
+            var client = new GitHubClient(new ProductHeaderValue("nuget-documentation-validator"))
+            {
+                Credentials = new Credentials(cred)
+            };
 
-            Console.WriteLine($"Checking {allLogCodes.Count} log codes for documentation.");
+            List<NuGetLogCode> undocumentedLogCodes = await GetUndocumentedLogCodes(allLogCodes, httpClient);
+            Console.WriteLine($"Undocumented Log Codes Count {undocumentedLogCodes.Count}");
+
+            Dictionary<NuGetLogCode, List<string>> logCodeToIssue =
+                await GetIssuesForLogCodeAsync(client, undocumentedLogCodes);
+            PrintIssues(logCodeToIssue);
+
+            bool shouldCreateIssues = false;
+            _ = args.Length >= 2 ? bool.TryParse(args[1], out shouldCreateIssues) : false;
+
+            if (shouldCreateIssues)
+            {
+                Console.WriteLine($"Creating issues for log codes without one.");
+                foreach (var logCodeIssueKVP in logCodeToIssue)
```

Comment:
Just general code-quality comment, not specific to small tools like this one. I like to try to use good coding practises in small tools, so it becomes habit and I do it automatically in our production code.

Consider moving this loop (actually everything inside the if statement) to a method. There are times when I'm debugging code, and I find it easier when a single step-over finishes a loop, rather than having to press F10 a whole lot of times, or move the cursor to another location and then use "run to cursor".

Also, if an exception happened (particularly in release builds without pdbs), then it's much easier to isolate when you have small methods, so the stack trace can point you to a small bit of code. NuGet.Client really fails at this 😞